runner/runner.go

```diff
@@ -0,0 +1,198 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create the consul template runner")
+	}
+
+	ctx, ctxCancel := context.WithCancel(context.Background())
+
+	runner := &Runner{
+		cfg:            cfg,
+		ctx:            ctx,
+		ctxCancel:      ctxCancel,
+		templateRunner: templateRunner,
+	}
+
+	return runner, nil
+}
+
+func DefaultConfig() (Config, error) {
+	return Config{
+		AraServer:              "http://127.0.0.1:8000",
+		ConsulAddr:             "127.0.0.1:8500",
+		Interval:               5 * time.Minute,
+		AnsibleFolder:          "/usr/etc/trento",
+		ConsulTemplateLogLevel: "info",
+	}, nil
+}
+
+func (c *Runner) Start() error {
+	var wg, renderedWg sync.WaitGroup
+
+	wg.Add(1)
+	renderedWg.Add(1)
+	go func(wg, renderedWg *sync.WaitGroup) {
+		log.Println("Starting consul-template loop...")
+		defer wg.Done()
+		c.startConsulTemplate(renderedWg)
+		log.Println("consul-template loop stopped.")
+	}(&wg, &renderedWg)
+
+	// Wait until the ansible inventory file is rendered
+	renderedWg.Wait()
+
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		log.Println("Starting the runner loop...")
+		defer wg.Done()
+		c.startRunnerTicker()
+		log.Println("Runner loop stopped.")
+	}(&wg)
+
+	wg.Wait()
+
+	return nil
+}
+
+func (c *Runner) Stop() {
+	c.ctxCancel()
+}
+
+func createAnsibleFiles(folder string) error {
+	log.Infof("Creating the ansible file structure in %s", folder)
+	// Clean the folder if it stores old files
+	ansibleFolder := path.Join(folder, "ansible")
+	err := os.RemoveAll(ansibleFolder)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	err = os.MkdirAll(ansibleFolder, 0644)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	// Create the ansible file structure from the FS
+	err = fs.WalkDir(ansibleFS, "ansible", func(fileName string, dir fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if !dir.IsDir() {
+			content, err := ansibleFS.ReadFile(fileName)
+			if err != nil {
+				log.Errorf("Error reading file %s", fileName)
+				return err
+			}
+			f, err := os.Create(path.Join(folder, fileName))
+			if err != nil {
+				log.Errorf("Error creating file %s", fileName)
+				return err
+			}
+			fmt.Fprintf(f, "%s", content)
+		} else {
+			os.Mkdir(path.Join(folder, fileName), 0644)
+		}
+		return nil
+	})
+
+	if err != nil {
+		log.Errorf("An error ocurred during the ansible file structure creation: %s", err)
+		return err
+	}
+
+	log.Info("Ansible file structure successfully created")
+
+	return nil
+}
+
+func (c *Runner) startRunnerTicker() {
+	err := createAnsibleFiles(c.cfg.AnsibleFolder)
+	if err != nil {
+		return
+	}
+
+	ansibleRunner, err := NewAnsibleRunner(
+		path.Join(c.cfg.AnsibleFolder, AnsibleMain),
+		path.Join(c.cfg.AnsibleFolder, ansibleHostFile))
+	if err != nil {
+		return
+	}
+
+	err = ansibleRunner.LoadAraPlugins()
+	if err != nil {
+		return
+	}
+
+	ansibleRunner.SetAraServer(c.cfg.AraServer)
+
+	tick := func() {
+		if !ansibleRunner.isAraServerUp() {
```

Comment:
Try to add the consul-template execution here, as `run-once`

Reply:
Changed in: https://github.com/trento-project/trento/pull/150/commits/6972c8eee957f196d80f0127d95fb4f3526bd79e